paasta_tools/iptables.py

```diff
@@ -0,0 +1,113 @@
+"""iptables helper functions.
+
+Unlike the `firewall` module, these functions know nothing about PaaSTA and
+could effectively be a third-party library. They just make working with
+iptables a little bit easier.
```

Comment:
I didn't originally, but I did just now. Of the ones that look to be maintained/reasonably popular, the one that looks most interesting to me is https://github.com/ldx/python-iptables.

Unlike my code, it provides Python bindings for working with the C interface to iptables. That would almost certainly be more performant and may make some things easier (e.g. maybe we no longer have to worry about parsing the rules, since it provides access to the structs used by the C API).

I'll update my review with the changes so far, and look into using this library.